### PR TITLE
OCaml port!

### PR DESCRIPTION
**Pull request by ajbt200128**
Technically the README needs to be updated, but at least for the purposes of getting this in the hands of users we can do that later
### Security

- [X] Change has security implications (if so, ping the security team)

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -21,27 +24,46 @@ export function registerCommands(
...
     }
   });
 
-  vscode.commands.registerCommand("semgrep.scanWorkspace", async () => {
-    await client.sendNotification(scanWorkspace);
+  vscode.commands.registerCommand("semgrep.loginNudge", async () => {
+    env.logger.log("logged in: " + env.loggedIn);
+    if (!env.loggedIn && env.showNudges) {
+      const resp = await vscode.window.showInformationMessage(
+        "Login to enable additional proprietary Semgrep Registry rules and running custom policies from Semgrep Code",
```
Comment (underyx) [inline src/commands.ts:58]: do we actually have login-only rules today?

Comment (ajbt200128) [reply src/commands.ts:58]: SSC rules, and I was under the impression that "pro rules" were login only

Comment (underyx) [inline src/commands.ts:58]:
```suggestion
        "Sign in to use your team's shared Semgrep rule configuration",
```

File: src/commands.ts [hunk 2]
```diff
@@ -21,27 +24,46 @@ export function registerCommands(
...
+        "Login",
```
Comment (underyx) [inline src/commands.ts:59]:
```suggestion
        "Sign in",
```

Just using the same words as on https://semgrep.dev/login

File: package.json [hunk 3]
```diff
@@ -29,27 +29,35 @@
...
       },
       {
-        "command": "semgrep.scanWorkspace",
-        "title": "Semgrep: Scan workspace"
+        "command": "semgrep.scanWorkspaceFull",
+        "title": "Semgrep: Scan all files in workspace"
+      },
+      {
+        "command": "semgrep.logout",
+        "title": "Semgrep: Logout"
```
Comment (underyx) [inline package.json:56]:
```suggestion
        "title": "Semgrep: Sign out"
```

same words as on semgrep.dev

File: package.json [hunk 4]
```diff
@@ -29,27 +29,35 @@
...
+          "fontPath": "./semgrep-icon.woff",
+          "fontCharacter": "A"
         }
-      ]
+      }
     },
     "commands": [
       {
         "command": "semgrep.login",
         "title": "Semgrep: Login"
```
Comment (underyx) [inline package.json:44]:
```suggestion
        "title": "Semgrep: Sign in"
```

```suggestion
        "title": "Semgrep: Sign out"
```

same words as on semgrep.dev

**Review by underyx — APPROVED** [review]

File: src/commands.ts (hunk 1 above)
Comment (p4p3r) [reply src/commands.ts:58]: Correct, Pro rules are for paid users only, hence available only after logging in.

**Pull request merged**